vinscant/src/main.rs

```diff
@@ -0,0 +1,232 @@
+    // implemented by esp-idf-sys might not link properly. See https://github.com/esp-rs/esp-idf-template/issues/71
+    esp_idf_svc::sys::link_patches();
+
+    // Bind the log crate to the ESP Logging facilities
+    esp_idf_svc::log::EspLogger::initialize_default();
+
+    let peripherals = Peripherals::take().unwrap();
+    let sysloop = EspSystemEventLoop::take().unwrap();
+
+    // The constant `CONFIG` is auto-generated by `toml_config`.
+    let _wifi_thing = wifi::wifi(
+        CONFIG.wifi_ssid,
+        CONFIG.wifi_psk,
+        peripherals.modem,
+        sysloop,
+    );
+
+    let pins = peripherals.pins;
+
+    #[cfg(feature = "esp32s2")]
+    let scan_spi_device = SpiSingleDeviceDriver::new_single(
+        peripherals.spi2,
+        pins.gpio34.downgrade_output(), // SCK
+        pins.gpio35.downgrade_output(), // MOSI
+        Some(pins.gpio36.downgrade_input()), // MISO
+        Some(pins.gpio33.downgrade_output()), // CS/SDA
+        &spi::config::DriverConfig::new(),
+        &spi::config::Config::new(),
+    )
+    .unwrap();
+
+    #[cfg(feature = "esp32")]
+    let scan_spi_device = SpiSingleDeviceDriver::new_single(
+        peripherals.spi2,
+        pins.gpio0.downgrade_output(), // SCK
+        pins.gpio4.downgrade_output(), // MOSI
+        Some(pins.gpio27.downgrade_input()), // MISO
+        Some(pins.gpio13.downgrade_output()), // CS/SDA
+        &spi::config::DriverConfig::new(),
+        &spi::config::Config::new(),
+    )
+    .unwrap();
+
+    let scan_interface = SpiInterface::new(scan_spi_device);
+    let mut scanner = Mfrc522::new(scan_interface).init().unwrap();
+
+    #[cfg(feature = "esp32s2")]
+    let led_pin = pins.gpio17;
+    #[cfg(feature = "esp32")]
+    let led_pin = pins.gpio5;
+    let channel = peripherals.rmt.channel0;
+    let led_strip = LedPixelEsp32Rmt::<RGB8, LedPixelColorGrb24>::new(channel, led_pin).unwrap();
+
+    let mut status_notifier = StatusNotifier {
+        led_strip,
+        leds: 8,
+        idle_effect: Box::new(PingPong::new(8, vec![Srgb::new(0xff, 0x7f, 0x00)])),
+        //buzzer: Buzzer::new(&peripherals)
+    };
+
+    let mut last_uid = hex::encode([0_u8]);
+    let mut last_time = 0;
+    // buzzer testing {{{
+    if false {
+        let mut timer_driver = LedcTimerDriver::new(
+            peripherals.ledc.timer0,
+            &config::TimerConfig::new().frequency(100.into()),
+        )
+        .unwrap();
+        #[cfg(feature = "esp32s2")]
+        let mut channel = LedcDriver::new(
+            peripherals.ledc.channel0,
+            &timer_driver,
+            pins.gpio37.downgrade_output(),
+        )
+        .unwrap();
+
+        #[cfg(feature = "esp32")]
+        let mut channel = LedcDriver::new(
+            peripherals.ledc.channel0,
+            &timer_driver,
+            pins.gpio19.downgrade_output(),
+        )
+        .unwrap();
+        for numerator in [1, 2, 3, 4, 5, 6].iter().cycle() {
```

Comment:
```suggestion
        for numerator in (1..=6).cycle() {
```
use the range syntax